src/domain.rs

```diff
@@ -0,0 +1,31 @@
+/// Configuration for domain checks to be performed on either the `Host`
+/// or the `Origin` header.
+#[derive(Debug)]
+pub enum DomainCheck<Domain = String> {
```

Comment:
How about `Policy`? If we rename the module `access_control` we get the fairly readable `access_control::Policy`?